Investigate KafkaDevServicesDisabledTest failure test

Currently if  ```KafkaDevServicesDisabledTest``` is running after ```KafkaDevServicesEabledTest```, it is failing with
```
[ERROR] camelKafkaComponentBrokersConfigurationNull  Time elapsed: 0.014 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <null> but was: <PLAINTEXT://localhost:33066>
```
It looks like the kafka dev service is avaible during all the test phase regardless of the test is restarted even if ```quarkus.kafka.devservices.enabled=false``` is configed.

It might be a bug somewhere in Quarkus DevService.

--- a/extensions/kafka/deployment/src/main/java/org/apache/camel/quarkus/component/kafka/deployment/KafkaProcessor.java
+++ b/extensions/kafka/deployment/src/main/java/org/apache/camel/quarkus/component/kafka/deployment/KafkaProcessor.java
@@ -24,8 +24,8 @@ import io.quarkus.deployment.Capability;
 import io.quarkus.deployment.IsNormal;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
-import io.quarkus.deployment.builditem.DevServicesConfigResultBuildItem;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.RunTimeConfigurationDefaultBuildItem;
 import io.quarkus.deployment.dev.devservices.GlobalDevServicesConfig;
 import io.quarkus.kafka.client.deployment.DevServicesKafkaBrokerBuildItem;
 import io.quarkus.kafka.client.deployment.KafkaBuildTimeConfig;
@@ -55,13 +55,13 @@ class KafkaProcessor {
     public void configureKafkaComponentForDevServices(
             DevServicesKafkaBrokerBuildItem kafkaBrokerBuildItem,
             KafkaBuildTimeConfig kafkaBuildTimeConfig,
-            BuildProducer<DevServicesConfigResultBuildItem> devServiceConfig) {
+            BuildProducer<RunTimeConfigurationDefaultBuildItem> runTimeConfig) {
 
         Config config = ConfigProvider.getConfig();
         Optional<String> brokers = config.getOptionalValue(CAMEL_KAFKA_BROKERS, String.class);
 
         if (brokers.isEmpty() && kafkaBuildTimeConfig.devservices.enabled.orElse(true)) {
-            devServiceConfig.produce(new DevServicesConfigResultBuildItem(CAMEL_KAFKA_BROKERS,
+            runTimeConfig.produce(new RunTimeConfigurationDefaultBuildItem(CAMEL_KAFKA_BROKERS,
                     kafkaBrokerBuildItem.getBootstrapServers()));
         }
     }

--- a/extensions/kafka/deployment/src/test/java/org/apache/camel/quarkus/component/kafka/deployment/KafkaDevServicesDisabledTest.java
+++ b/extensions/kafka/deployment/src/test/java/org/apache/camel/quarkus/component/kafka/deployment/KafkaDevServicesDisabledTest.java
@@ -34,8 +34,7 @@ public class KafkaDevServicesDisabledTest {
     @RegisterExtension
     static final QuarkusUnitTest CONFIG = new QuarkusUnitTest()
             .withConfigurationResource("application-configuration-devservices-disabled.properties")
-            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class))
-            .overrideConfigKey("camel.component.kafka.brokers", "");
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class));
 
     @Inject
     CamelContext context;

